### DISCUSSION TITLE
Language support?

### DISCUSSION THREAD

**Discussion by Felderburg** [bcholmes/StarTrek2d20#84] [Ideas]
A user asked about other languages at Modiphius (https://forums.modiphius.com/t/star-trek-adventures-online-character-generator/94/385). The reply linked the google translated site (https://sta-bcholmes-org.translate.goog/?_x_tr_sl=en&_x_tr_tl=es (it's not clear to me why all the stuff at the end of that url is necessary, but it is; note that the original link at modiphius has the 'language > other language' showing up in Spanish)).

The google translate version seems to work pretty well, but I thought it might be useful to start a discussion about potentials for language support. Obviously it would take 2 big things: lots of dev effort, and people willing to translate. The fairly rapid addition of pages, such as the new option for starship build types, implies to me that there could be a landing page with language choice buttons (and a link to the google translate version for languages not included).

Given the google translate link, I'm not sure this is necessary, especially given the size of the dev team, but thought I'd put it up as discussion on the git.

**Comment by Felderburg** [comment]
So this is well underway now, but how do I test it?

**Comment by bcholmes** [comment]
National language support is driven by the browser's language. Most browsers give you the ability to change the language (e.g., in Firefox, go to "Preferences", and under the "General" section there's a heading for "Language and Appearance". Chrome has something similar). If you browser announces its language as, say, German, the site will appear in German.

**Reply by kaktoforum** [reply]
And what shall I do if my default language is English and I need to generate the character in other language?
Personally hate to have translated OS and using EN as environmental language for all my devices.
That doesn't mean that I can't use other two languages for specific applications and different occasions.
So if I could use some sort of visible switch in addition, it could be helpful.

**Reply by kaktoforum** [reply]
Meanwhile found a workaround
In FireFox it's F12 (FireBug if not mistaken)
Currently de/en/es/fr (waiting for ru) and automatically becoming en for unsupported.

![2024-02-14_19-54](https://github.com/bcholmes/StarTrek2d20/assets/77748199/fd29db7a-dcf0-4b91-8ed1-9110b7bd3f7e)

**Comment by Felderburg** [comment]
Well I tried changing things around, it seemed to work alright. One oddity is that even after I removed German from my browser, I had to delete the cookie to see the site in English again.

After I had removed English entirely from my browser (which I had to do because Chrome couldn't figure out that I wanted the UI *and translations* in German) it still showed German even after I clicked the "switch back to English" button on the front page "popup". I'm not sure if that's a Chrome thing or what.

Edit: Probably a chrome thing since it's still trying to "translate" everything I had open into English instead of German. Very weird behavior (I guess it's not that weird if google popped a cookie into every tab I had open that said 'hey translate this to English from German' but it is annoying that it persists after I made the settings changes.).
